src/morphosource/search.py

```diff
@@ -17,18 +17,47 @@ def _get(obj, name, unlist=True):
     return value
 
 
+class ObjectTypes(object):
+    BIOLOGICAL_SPECIMEN = "Biological Specimen"
+    CULTURAL_HERITAGE = "Cultural Heritage Object"
+
+
 class Media(object):
     def __init__(self, data):
         self.id = _get(data, "id")
         self.title = _get(data, "title")
         self.media_type = _get(data, "media_type")
         self.visibility = _get(data, "visibility")
+        self.physical_object_id = _get(data, "physical_object_id")
         self.data = data
 
     def download_bundle(self, path, download_config):
         download_media_bundle(media_id=self.id, path=path, download_config=download_config)
 
 
+class PhysicalObject(object):
+    def __init__(self, data):
+        self.id = _get(data, "id")
+        self.title = _get(data, "title")
+        self.type = _get(data, "type")
+        self.taxonomy = _get(data, "taxonomy")
+        self.data = data
+
+    def get_media_ary(self, open_visibility_only=False):
+        results = []
+        media_search_results = search_media(query=self.id)
+        for media in media_search_results.items:
+            if self.id == media.physical_object_id:
+                if open_visibility_only:
+                    # NOTE: The value returned in visiblity is different than the facet 
```

Comment:
What does this mean?

Reply:
In the MorphoSource UI there is a `Publication` facet that can be filtered by the values `Open Download` and `Restricted Download`. The JSON payload returned from the MorphoSource API includes a `visibility` field that contains the same data but uses different values: `open` and `restricted_download`. So the `if` below filters for "open".

Reply:
This is for open access downloads. After a second thought I could move the filtering further up and use the facet.